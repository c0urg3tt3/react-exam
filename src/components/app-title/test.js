import React from 'react'
import { shallow } from 'enzyme'

import AppTitle from './component'

describe('[component] AppTitle', () => {
  it('should render component', () => {
    const wrapper = shallow(<AppTitle/>)

    expect(wrapper.exists()).toEqual(true)
    expect(wrapper.is('h1')).toEqual(true)
    expect(wrapper.hasClass('app-title')).toEqual(true)
    expect(wrapper.text()).toBe('Jedi List')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('with custom title', () => {
    const wrapper = shallow(<AppTitle title={"Sith List"}/>)

    expect(wrapper.exists()).toEqual(true)
    expect(wrapper.is('h1')).toEqual(true)
    expect(wrapper.hasClass('app-title')).toEqual(true)
    expect(wrapper.text()).toBe('Sith List')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
