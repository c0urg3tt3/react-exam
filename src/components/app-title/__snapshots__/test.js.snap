// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] AppTitle should render component 1`] = `"<h1 class=\\"app-title\\">Jedi List</h1>"`;

exports[`[component] AppTitle with custom title 1`] = `"<h1 class=\\"app-title\\">Sith List</h1>"`;
